### PR TITLE
Check status of multipathd service in attach detach scripts

### PR DESCRIPTION
**Pull request by IBMCaoBiao**
Only modprobe dm_multipath is not enough when executing attach detach
scripts, should make sure multipathd service is active.

Signed-off-by: CaoBiao <bjcb@cn.ibm.com>

### CODE REVIEW

**Comment by IBMCaoBiao** [comment]
# test results

## 1. stop multipathd server in target vm
```
[root@rhel86-scsi-capture-1 ~]# systemctl stop multipathd
Warning: Stopping multipathd.service, but it can still be activated by:
  multipathd.socket
[root@rhel86-scsi-capture-1 ~]# 
[root@rhel86-scsi-capture-1 ~]# systemctl status multipathd
● multipathd.service - Device-Mapper Multipath Device Controller
   Loaded: loaded (/usr/lib/systemd/system/multipathd.service; enabled; vendor preset: enabled)
   Active: inactive (dead) since Tue 2022-08-16 05:40:32 CDT; 8s ago
  Process: 909 ExecStart=/sbin/multipathd -d -s (code=exited, status=0/SUCCESS)
  Process: 908 ExecStartPre=/sbin/multipath -A (code=exited, status=0/SUCCESS)
  Process: 906 ExecStartPre=/sbin/modprobe -a scsi_dh_alua scsi_dh_emc scsi_dh_rdac dm-multipath (code=exited, status=0/SUCCESS)
 Main PID: 909 (code=exited, status=0/SUCCESS)
   Status: "shutdown"

Aug 15 23:29:22 rhel86-scsi multipathd[909]: mpatha: sdc - tur checker reports path is up
Aug 15 23:29:22 rhel86-scsi multipathd[909]: 8:32: reinstated
Aug 15 23:29:22 rhel86-scsi multipathd[909]: mpatha: remaining active paths: 4
Aug 15 23:29:24 rhel86-scsi multipathd[909]: reconfigure (operator)
Aug 15 23:29:25 rhel86-scsi multipathd[909]: mpatha: load table [0 20971520 multipath 1 queue_if_no_path 1 alua 2 1 service-time 0 2 1 8:32 1 8:16 1 service-time 0 2 1 8:48>
Aug 16 05:40:27 rhel86-scsi-capture-1 systemd[1]: Stopping Device-Mapper Multipath Device Controller...
Aug 16 05:40:27 rhel86-scsi-capture-1 multipathd[909]: exit (signal)
Aug 16 05:40:27 rhel86-scsi-capture-1 multipathd[909]: --------shut down-------
Aug 16 05:40:32 rhel86-scsi-capture-1 systemd[1]: multipathd.service: Succeeded.
Aug 16 05:40:32 rhel86-scsi-capture-1 systemd[1]: Stopped Device-Mapper Multipath Device Controller.
[root@rhel86-scsi-capture-1 ~]# 
```

## 2. detach a volume
```
Volume [rhel86_scsi_capture_1-data-123](https://172.26.4.29/icic/index.html#t=storageVolumes.volume_details&volId=f4e47b0e-c100-4888-af87-1d1cf4c6dcb2) could not be detached from virtual machine [rhel86_scsi_capture_1](https://172.26.4.29/icic/index.html#t=virtualMachines.vm_details&vmId=37ce6d1f-bdcb-4140-8292-690d5c942b4b). Error message: zVM Cloud Connector request failed: {'overallRC': 500, 'modID': 100, 'rc': 500, 'rs': 1, 'errmsg': "Unexpected internal error in ZVM SDK, error: (127.0.0.1:46156) SDK server got exception without results defined, error: z/VM SDK error: detach failed with error:Failed to detach volume from instance BJCB001F with reason detach scripts execution on fcp ['1b12', '1c12'] in the target virtual machine BJCB001F failed with unknow reason, exit code is: 3", 'output': ''}
```

the error log in VM:
```
Aug 16 05:41:12 rhel86-scsi-capture-1 zvmguestconfigure[11314]: zvmguestconfigure is starting#012Aug 16 05:41:12 rhel86-scsi-capture-1 zvmguestconfigure[11318]: WARNING: Device[0.0.000c] is already  online#012Aug 16 05:41:12 rhel86-scsi-capture-1 zvmguestconfigure[11314]: Downloading record 0008: atvol.sh
Aug 16 05:41:12 rhel86-scsi-capture-1 iucvserv[1009]: result length=1157, send message length=1022,#012 Aug 16 05:41:12 rhel86-scsi-capture-1 zvmguestconfigure[11342]: Enter RHEL8 attach script with parameters: FCP list:1b12 1c12, INPUT WWPNS:0x5005076306031388 0x5005076306035388 0x5005076306039388 0x500507630603d388 0x5005076306081388 0x5005076306085388 0x5005076306089388 0x500507630608d388 0x5005076306101388 0x5005076306105388 0x5005076306109388 0x500507630610d388 0x5005076306131388 0x5005076306135388 0x5005076306139388 0x500507630613d388 0x5005076306181388 0x5005076306185388 0x5005076306189388 0x500507630618d388 0x50050763061b1388 0x50050763061b5388 0x50050763061b9388 0x50050763061bd388, LUN:0x4017401f00000000, target_filename:vdb#012Aug 16 05:41:12 rhel86-scsi-capture-1 zvmguestconfigure[11342]: Checking status of multipathd service ...#012Aug 16 05:41:12 rhel86-scsi-capture-1 zvmguestconfigure[11342]: Attach script terminated because multipathd service is not active!#012Aug 16 05:41:12 rhel86-scsi-capture-1 zvmguestconfigure[11314]: zvmguestconfigure has successfully processed the reader files with exit code: 3.
Aug 16 05:41:12 rhel86-scsi-capture-1 iucvserv[1009]: result length=384, send message length=384,#012 Aug 16 05:41:12 rhel86-scsi-capture-1 systemd[1]: zvmguestconfigure.service: Main process exited, code=exited, status=3/NOTIMPLEMENTED#012Aug 16 05:41:12 rhel86-scsi-capture-1 systemd[1]: zvmguestconfigure.service: Failed with result 'exit-code'.#012Aug 16 05:41:12 rhel86-scsi-capture-1 systemd[1]: Failed to start Activation engine for configuring zLinux os when it start up.#012iucvcmdrc=3

```

**Comment by IBMCaoBiao** [comment]
# test results: attach volume

**Comment by dongyanyang** [comment]
@IBMCaoBiao why the error msg is still "failed with unknown reason, exit code is 3"? avoid the "unknown reason" if we are sure exit code 3 represents the multipathd service error.

File: zvmsdk/volumeops/templates/rhel7_attach_volume.j2 [hunk 1]
```diff
@@ -7,6 +7,29 @@ target_filename="{{ target_filename }}"
...
+    echo "dm-multipath mode enabled successfully"
+else
+    echo "dm-multipath mode is already enabled"
+fi
+echo "Checking status of multipathd service ..."
+check_multipathd_service=`systemctl is-active --quiet multipathd`
+exit_code=$?
+if [[ $exit_code != 0 ]]; then
+    echo "Attach script terminated because multipathd service is not active!"
+    exit $exit_code
```
Comment (dongyanyang) [inline zvmsdk/volumeops/templates/rhel7_attach_volume.j2:30]: define our own exit_code for multipathd error, and log the exit_code of line 27 into line 29.

**Pull request merged**